pysteps/nowcasts/anvil.py

```diff
@@ -0,0 +1,370 @@
+    n_cascade_levels : int, optional
+        The number of cascade levels to use.
+    extrap_method : str, optional
+        Name of the extrapolation method to use. See the documentation of
+        pysteps.extrapolation.interface.
+    ar_order : int, optional
+        The order of the autoregressive model to use.
+    ar_window_radius : int, optional
+        The radius of the window to use for determining the parameters of the
+        autoregressive model.
+    r_vil_window_radius : int, optional
+        The radius of the window to use for determining the R(VIL) relation.
+        Applicable if rainrate is not None.
+    fft_method : str, optional
+        A string defining the FFT method to use (see utils.fft.get_method).
+        Defaults to 'numpy' for compatibility reasons. If pyFFTW is installed,
+        the recommended method is 'pyfftw'.
+    num_workers : int, optional
+        The number of workers to use for parallel computation. Applicable if
+        dask is installed or pyFFTW is used for computing the FFT.
+        When num_workers>1, it is advisable to disable OpenMP by setting
+        the environment variable OMP_NUM_THREADS to 1.
+        This avoids slowdown caused by too many simultaneous threads.
+    extrap_kwargs : dict, optional
+        Optional dictionary containing keyword arguments for the extrapolation
+        method. See the documentation of pysteps.extrapolation.
+    filter_kwargs : dict, optional
+        Optional dictionary containing keyword arguments for the filter method.
+        See the documentation of pysteps.cascade.bandpass_filters.py.
+    measure_time : bool, optional
+        If True, measure, print and return the computation time.
+
+    Returns
+    -------
+    out : ndarray
+        A three-dimensional array of shape (n_timesteps,m,n) containing a time
+        series of forecast precipitation fields. The time series starts from
+        t0+timestep, where timestep is taken from the input VIL/rain rate
+        fields. If measure_time is True, the return value is a three-element
+        tuple containing the nowcast array, the initialization time of the
+        nowcast generator and the time used in the main loop (seconds).
+
+    References
+    ----------
+    :cite:`PCHL2020`
+
+    Notes
+    -----
+    The original ANVIL method developed in :cite:`PCHL2020` uses VIL as the
+    input quantity. The forecast model is, however, more general and can take
+    any two-dimensional input field.
+    """
+    if len(vil.shape) != 3:
+        raise ValueError("vil.shape = %s, but a three-dimensional array expected" % str(vil.shape))
+
+    if rainrate is not None:
+        if len(rainrate.shape) != 2:
+            raise ValueError("rainrate.shape = %s, but a two-dimensional array expected" % str(rainrate.shape))
+
+    if vil.shape[0] != ar_order + 2:
+        raise ValueError("vil.shape[0] = %d, but vil.shape[0] = ar_order = %d required" % (vil.shape[0], ar_order))
```

Comment:
Looks like something is wrong here: the if statement checks whether `vil.shape[0] != ar_order + 2`, but the error message says that `vil.shape[0]` must be equal to `ar_order`.